src/uct/ib/rdmacm/rdmacm_iface.c

```diff
@@ -377,19 +368,9 @@ uct_rdmacm_iface_process_event(uct_rdmacm_iface_t *iface,
         /* Client - connection is ready */
         ucs_assert(!iface->is_server);
         ret_flags |= UCT_RDMACM_PROCESS_EVENT_DESTROY_CM_ID_FLAG;
-        if (ep) {
+        if (ep != NULL) {
             ep->status = UCS_OK;
-            ucs_queue_head_init(&completed_ops);
-            ucs_queue_for_each_safe(op, iter, &ep->ops, queue_elem) {
-                if (--op->user_comp->count == 0) {
-                    op->user_comp->func(op->user_comp, UCS_OK);
-                    ucs_queue_del_iter(&ep->ops, iter);
-                    ucs_queue_push(&completed_ops, &op->queue_elem);
-                }
-            }
-            ucs_queue_for_each_extract(op, &completed_ops, queue_elem, 1) {
-                ucs_free(op);
-            }
+            uct_rdmacm_ep_purge_outstanding(ep, UCS_OK);
```

Comment:
i would rename this function to uct_rdmacm_ep_invoke_completions() or similar, because "purge" somewhat implies it's only called from cleanup/error flow, while it's actually called also from "good" flow